external/bart/bart/sched/functions.py

```diff
@@ -139,6 +139,25 @@ def sanitize_asymmetry(series, window=None):
                 if window:
                     series.index.values[-1] = window[1]
 
+            # Remove repeated entries - which could happen if a task switch in
+            # then immediately switches out; ie: time stamp is exactly the same
+            n = 0
+            next = n + 1
+            while next < len(series):
+                if not series.values[n]:
+                    n = next
+                    next = next + 1
+                    continue
+
+                while not series.values[next]:
```

Comment:
This may be faster depending on how indexing is actually implemented as it increments without loosing the current position (something like a linked list would benefit a lot from that):
```
for i, val in enumerate(series.values[next:]):
    if val:
        break
next = next + i
```

Reply:
Okay I'll try that